From 6d1a236b2c16a4109cfcad20bbeb0922192d92c8 Mon Sep 17 00:00:00 2001
Message-Id: <6d1a236b2c16a4109cfcad20bbeb0922192d92c8.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Wed, 28 Jan 2015 21:49:42 +0530
Subject: [PATCH 54/95] MUST_REBASE [VPG]: video/adf/intel: Add support for
 punit reg read/write

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5594
Change-Id: I305de99591f84245efe942c0bd5ecb8ed151dfd9
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 .../adf/intel/include/core/vlv/vlv_dc_config.h     |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h b/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
index 524c669..8153424 100644
--- a/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
+++ b/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
@@ -136,6 +136,21 @@ static inline void vlv_cck_write(u32 reg, u32 val)
 	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_WRITE, IOSF_PORT_CCK,
 					reg, &val);
 }
+
+static inline u32 vlv_punit_read(u32 reg)
+{
+	u32 val;
+	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_READ, IOSF_PORT_PUNIT,
+			reg, &val);
+	return val;
+}
+
+static inline void vlv_punit_write(u32 reg, u32 val)
+{
+	intel_adf_pci_sideband_rw(INTEL_SIDEBAND_REG_WRITE, IOSF_PORT_PUNIT,
+			reg, &val);
+}
+
 struct vlv_pipeline {
 	struct intel_pipeline base;
 	struct vlv_dpst *dpst;
-- 
1.7.9.5

